test/rt-test.py

```diff
@@ -90,33 +76,38 @@ def cleanup():
     # The list of log statements to look for before arch init is called +
     # All log statements to look for, including the arch-specific ones +
     # The list of log statements to look for after arch init is called
-    cases = get_pre_archinit_cases() + arch_module.get_test_cases(TestCase) + get_post_archinit_cases()
-
-    if len(cases) > 0:
-        proc = subprocess.Popen(zig_path + " build run -Drt-test=true", stdout=subprocess.PIPE, shell=True, preexec_fn=os.setsid)
-        atexit.register(cleanup)
-        case_idx = 0
-        read_thread = threading.Thread(target=read_messages, args=(proc,))
-        read_thread.daemon = True
-        read_thread.start()
-        # Go through the cases
-        while case_idx < len(cases):
-            case = cases[case_idx]
-            expected_idx = 0
-            # Go through the expected log messages
-            while expected_idx < len(case.expected):
-                e = case.prefix + case.expected[expected_idx]
-                try:
-                    line = msg_queue.get(block=True, timeout=5)
-                except queue.Empty:
-                    failure("Timed out waiting for '%s'" %(e))
-                line = line.strip()
-                pattern = re.compile(e)
-                # Pass if the line matches the expected pattern, else fail
-                if pattern.fullmatch(line):
-                    test_pass(case, e, expected_idx, line)
-                else:
-                    test_failure(case, e, expected_idx, line)
-                expected_idx += 1
-            case_idx += 1
+    cases = get_pre_archinit_cases() + arch_module.get_test_cases() + get_post_archinit_cases()
+
+    proc = subprocess.Popen(zig_path + " build run -Drt-test=true", stdout=subprocess.PIPE, shell=True, preexec_fn=os.setsid)
+    atexit.register(cleanup)
+    
+    read_thread = threading.Thread(target=read_messages, args=(proc,))
+    read_thread.daemon = True
+    read_thread.start()
+
+    while cases:
+        try:
+            line = msg_queue.get(block=True, timeout=5)
+        except queue.Empty:
+            print("Missing cases: " + cases)
+            failure("Timed out")
+        
+        line = line.strip()
+
+        # Print the line so can see what is going on
+        print(line)
+
+        # If there is a FAILURE message in the log, then fail the testing
+        if "FAILURE" in line:
+            failure("Test failed")
+        
+        if not cases:
```

Comment:
Why is this check necessary? Doesn't the loop run until the list is empty? I may be misinterpreting the python here :)

Reply:
Yeh, I can remove this